Feature/line sensor plugin

lucastrschneider opened the pull request.

lucastrschneider committed:
🔧 Add vscode config

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>

lucastrschneider committed:
✨ Add line sensor plugin

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>

lucastrschneider committed:
🔥 Remove unused link

FelipeGdM commented on package.xml:
```diff
@@ -9,10 +9,10 @@
-  <author>Thalles Carneiro</author>
-
-  <maintainer email="felipegmelo@usp.br" />
+  <maintainer email="felipegmelo@usp.br">Felipe Gomes de Melo</maintainer>
+  <maintainer email="lucas.haug@thunderatz.org">Thalles Carneiro</maintainer>
+  <maintainer email="thalles.carneiro@thunderatz.org">Lucas Haug</maintainer>
+  <maintainer email="lucas.schneider@thunderatz.org">Lucas Schneider</maintainer>
```
Bora aproveitar e colocar todo mundo

FelipeGdM commented on launch/gazebo.launch:
```diff
@@ -25,6 +25,7 @@ https://github.com/ros/urdf_sim_tutorial -->
   <include file="$(find gazebo_ros)/launch/empty_world.launch">
     <arg name="debug" value="$(arg debug)" />
     <arg name="gui" value="$(arg gui)" />
+    <arg name="verbose" value="true" />
```
Tem certeza que essa deve ser uma opção permanente?

lucastrschneider replied:
então, a unica coisa que ele imprime acho que é a mensagem que a gnt colocou na hora que o plugin é criado, pq só dava pra ver com as coisas do gazebo, usando o ros debug stream não ia, ai por isso deixamos.
mas agora que ta tudo funcionando, acho que tirar ou não tanto faz

FelipeGdM commented on CMakeLists.txt:
```diff
@@ -1,14 +1,38 @@
-	install(DIRECTORY ${dir}/
-		DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/${dir})
-endforeach(dir)
+cmake_minimum_required(VERSION 3.0.0)
```
O ROS noetic recomenda a versão 3.0.2, item 2 do [guia de migração](http://wiki.ros.org/noetic/Migration)

```suggestion
cmake_minimum_required(VERSION 3.0.2)
```

lucastrschneider committed:
♻️ Refactor model
- Create other coordinate systems for each sensor mesh, so that the sensor plugin position can be easily configured

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>
Co-authored-by: Thalles Carneiro <mr-coala@users.noreply.github.com>

lucastrschneider committed:
🚚 Rename line sensors links

lucastrschneider committed:
✨ Add line sensor macro

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>

lucastrschneider committed:
💄 Add robot namespace to line sensor topic

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>

lucastrschneider committed:
⚡️ Add update rate to line sensor

Co-authored-by: Lucas Haug <LucasHaug@users.noreply.github.com>

lucastrschneider committed:
💥 Add the 5 line sensors to the model

lucastrschneider committed:
Update cmake minimum required version

Co-authored-by: Felipe Gomes de Melo <felipe.gomes@thunderatz.org>

lucastrschneider committed:
🔥 Remove visual from line sensor

lucastrschneider commented:
Agora sim, 5 sensores de linha funcionais no modelo!

A gente mudou o nome padrão dos links dos sensores de linha de "sensorx_link" para "linex_link", para deixar mais claro que tipo de sensor que era, já que tem o ultrassonico também ~por isso que tem um monte de arquivo modificado na PR agr~

LucasHaug commented on models/modelo_carrinho/model.sdf:
```diff
@@ -104,12 +104,12 @@
           <mesh>
             <scale>1 1 1</scale>
-            <uri>model://modelo_carrinho/meshes/sensor1_link.stl</uri>
+            <uri>model://modelo_carrinho/meshes/line1_link.stl</uri>
```
A gente esqueceu de mudar os nomes dos meshes dentro dessa pasta de modelos

LucasHaug commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Plugin Gazebo para sensor de linha](https://trello.com/c/SHZVfLZ1/15-plugin-gazebo-para-sensor-de-linha)

LucasHaug committed:
👥 Add project members
